addons/point_of_sale/static/src/app/models/data_service.js

```diff
@@ -298,6 +297,17 @@ export class PosData extends Reactive {
                     });
             }
 
+            if (result?.["pos.order"]) {
+                // Ensure loaded orders are linked to the current session,
+                // this modification is done here because is the most low
+                // level method that is called when loading orders
+                for (const order of result["pos.order"]) {
+                    if (order.state === "draft") {
+                        order.session_id = odoo.pos_session_id;
+                    }
+                }
+            }
+
```

Comment:
This is very specific to "pos.order" and it should be placed somewhere else. `execute` is supposed to be agnostic.